src/java/main/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -1451,7 +1452,12 @@ public synchronized void restartLeaderElection(QuorumVerifier qvOLD, QuorumVerif
     }
 
     public String getNextDynamicConfigFilename() {
-        return configFilename + QuorumPeerConfig.nextDynamicConfigFileSuffix;
+        if (configFilename != null) {
```

Comment:
@shralex Turns out pretty trivial to guarantee a real configFileName in this case - update with one line change. Tested on apache jenkins the issue is fixed.

Note we still have configFileName as null for some other tests (such as FLE tests) but in those tests the intermediate reconfig file is not relevant so I'll just leave as they are.